swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md

```diff
@@ -296,3 +298,46 @@ $ bal graphql -i service.bal -o ./schema -s /starwars
 The `service` parameter specifies the base path of the Ballerina GraphQL service of which the schema needs to be generated. This generates the GraphQL schema for the Ballerina GraphQL service in the `service.bal` file of which the `service-base-path` is `/starwars`.
 
 >**Info:** For more information on the command, see [Ballerina to GraphQL](/learn/cli-documentation/graphql/#ballerina-to-graphql).
+
+## Service generation
+
+The GraphQL tool supports generating a Ballerina GraphQL service from a given GraphQL schema. Successfull execution will generate two files. One contains the service, and other contains the types needed. You can export the generated files into a given location. The tool can be used in the following ways to generate a Ballerina GraphQL service.
+
+### Generate a service with GraphQL object types represented in service class types 
+
+If you want to generate service class types for GraphQL object types in the GraphQL schema, use the following command to generate the service.
+
+```
+$ bal graphql [-i | --input] <ballerina-service-file-path> [-m | --mode] <operation-mode> [-o | --output <output-location>] 
+```
+
+For example,
+
+```
+$ bal graphql -i schema.graphql -m service -o ./service
+```
+
+Output flag is optional. If it is omitted, the generated files will be written to the current directory. 
+
+>**Info:** For more information on the command, see [Ballerina to GraphQL](/learn/cli-documentation/graphql/#ballerina-to-graphql).
+
+### Generate a service with GraphQL object types represented in record types
+
+If you want to generate record types for GraphQL object types in the GraphQL schema, use the following command to generate the service. 
+
+```
+$ bal graphql [-i | --input] <ballerina-service-file-path> [-m | --mode] <operation-mode> [-o | --output <output-location>] [-r | --use-records-for-objects]
+```
+
+For example, 
+   
+```
+$ bal graphql -i schema.graphql -m service -o ./service -r
+```
+
+Some GraphQL object types can't be represented with record types. Those are identified as below.
```

Comment:
```suggestion
Even if the `[-r | --use-records-for-objects]` flag is used, the following object types will be generated using service types because these types cannot be represented by Ballerina record types.
```